### ISSUE TITLE
Negative numbers in the `set_filters_with_python_expr` of Report API raise an error.

### ISSUE THREAD

**Issue by matinmoezzi** [wandb/wandb#6837]
### Describe the bug

The following code raises the `getattr()` error. This is taken from the example notebook provided by WandB for Report API 

```python
pg.runsets = [
    wr.Runset(ENTITY, PROJECT, name="Grouping", groupby=["encoder"]),
    wr.Runset(ENTITY, PROJECT, name="Filtering").set_filters_with_python_expr("encoder == 'resnet18' and loss < -0.05"),
    wr.Runset(ENTITY, PROJECT, name="Ordering", order=["+momentum", "-Name"]),
]
report.save()

```

```shell
Traceback (most recent call last):
  File "/home/rl-baselines3-zoo/wandb_report_pick.py", line 23, in <module>
    runset = wr.runset.Runset("OOO", "PPP", query=q).set_filters_with_python_expr("n_timesteps == -1")
  File "/home/.condaenvs/rlzoo/lib/python3.10/site-packages/wandb/apis/reports/runset.py", line 75, in set_filters_with_python_expr
    self.filters = self.pm_query_generator.python_to_mongo(expr)
  File "/home/.condaenvs/lib/python3.10/site-packages/wandb/apis/public.py", line 3657, in python_to_mongo
    values = [self._handle_compare(tree.body)]
  File "/home/.condaenvs/rlzoo/lib/python3.10/site-packages/wandb/apis/public.py", line 3587, in _handle_compare
    right = self._handle_fields(node.comparators[0])
  File "/home/.condaenvs/rlzoo/lib/python3.10/site-packages/wandb/apis/public.py", line 3596, in _handle_fields
    result = getattr(node, self.AST_FIELDS.get(type(node)))
TypeError: getattr(): attribute name must be string
```
I figured it out that the reason is that the Python `ast` package reads negative number as `ast.UnaryOp` not `ast.constant` which raise error because the `ast.UnaryOp` is not defined in wandb `AST_FIELDS`
I solved this issue by modifying the `reports.py` in the following way:
(1) Added `ast.UnaryOp: "operand",` to the `AST_FIELDS` of `PythonMongoishQueryGenerator`
(2) Added these lines in the `handle_compare(self, node)` function:        
```
        if node.comparators[0].__class__.__name__ == "UnaryOp":
            right = ast.literal_eval(node.comparators[0])
        else:
             right = self._handle_fields(node.comparators[0])
            
```

### Environment

WandB version: 0.16.2

OS: Ubuntu 20

Python version: 3.9

Versions of relevant libraries:

**Comment by fmamberti-wandb** [comment]
Hi @matinmoezzi, thank you for taking the time to report this and investigate what's causing it. We will repro internally, work on a fix and we will keep you posted on the progress.

**Comment by fmamberti-wandb** [comment]
Hello @matinmoezzi, I have been trying to reproduce this on our side, but it seems to work fine in Colab and locally. Would you mind sharing additional information regarding the environment setup you are seeing this error?

**Comment by fmamberti-wandb** [comment]
Hi @matinmoezzi - I wanted to follow up on this request. Please let us know if we can be of further assistance or if your issue has been resolved on your side.

**Comment by fmamberti-wandb** [comment]
Hi @matinmoezzi, since we have not heard back from you we are going to close this request. If you would like to re-open the conversation, please let us know!

**Comment by matinmoezzi** [comment]
Hi @fmamberti-wandb, Sorry for the late response. I was not around at work for a few days. I solved that issue by modifying the code. What information do you need?

**Comment by fmamberti-wandb** [comment]
Hey @matinmoezzi - when trying to reproduce the issue on [this Notebook](https://github.com/wandb/examples/blob/master/colabs/intro/Report_API_Quickstart.ipynb), I was able to run it without having to make any changes to the `wandb` package on version 0.16.3. Would you mind testing this and let me know if it also works for you?

**Comment by fmamberti-wandb** [comment]
Hi @matinmoezzi, I wanted to follow up on this request. Please let us know if we can be of further assistance or if your issue has been resolved for you with 0.16.3.

**Comment by fmamberti-wandb** [comment]
Hi @matinmoezzi, since we have not heard back from you we are going to close this request. If you would like to re-open the conversation, please let us know!
